File: typescript/index-families/src/examples.ts
/**
 * Example: List index families.
 *
 * Demonstrates how to fetch index families from the dedicated
 * index-families endpoint.
 *
 * Set ANERA_MARKETS_API_BASE_URL to your API origin (scheme + host, no trailing slash).
 */

import { getJson } from "@anera/shared-client";
import { run } from "@anera/shared-client/util";
import type { IndexFamily } from "./types.js";

async function getIndexFamilies(): Promise<IndexFamily[]> {
  return getJson<IndexFamily[]>("/api/v1/index-families");
}

async function main(): Promise<void> {
  console.log("Index Families");
  console.log("=".repeat(80));

  try {
    const families = await getIndexFamilies();
    if (families.length === 0) {
      console.log("No index families available");
    }
    for (const family of families) {
      console.log(`\n${family.family_name} (${family.family_id}):`);
      const description = family.family_description;
      console.log(`  Description: ${description.length > 100 ? `${description.substring(0, 100)}...` : description}`);
      console.log(`  Tickers: ${family.family_tickers.join(", ")}`);
      const primary = family.primary_index;
      if (primary) {
        const value = primary.index_value !== null ? ` ${primary.index_value.toFixed(2)}` : "";
        console.log(`  Primary index: ${primary.symbol} (value:${value}, trend: ${primary.trend})`);
      }
    }
  } catch (err) {
    const error = err as Error;
    console.error(`Error: ${error.message}`);
  }

  console.log("\n" + "=".repeat(80));
}

run(main);

File: python/index-families/examples.py
"""
Example: List index families.

Demonstrates how to fetch index families (family metadata, member tickers,
and primary index details) from the dedicated index-families endpoint.

Set ANERA_MARKETS_API_BASE_URL to your API origin (scheme + host, no trailing slash).
"""

from __future__ import annotations

import requests
from typing import Any

from shared.http import get_json


def get_index_families() -> list[dict[str, Any]]:
    """Get all index families from the dedicated index-families endpoint."""
    return get_json("/api/v1/index-families")


def main() -> None:
    print("Index Families")
    print("=" * 80)

    try:
        families = get_index_families()
        print(f"Found {len(families)} families")

        for family in families:
            print(f"\n{family['family_name']}:")
            print(f"  Family ID:   {family['family_id']}")
            description = family.get("family_description") or ""
            print(f"  Description: {description[:100]}{'..' if len(description) > 100 else ''}")
            tickers = family.get("family_tickers") or []
            print(f"  Tickers:     {', '.join(tickers)}")

            primary = family.get("primary_index")
            if primary is not None:
                value = primary.get("index_value")
                value_str = f"{value:.2f}" if value is not None else "N/A"
                print(f"  Primary index: {primary['symbol']}")
                print(f"    Value: {value_str}")
                print(f"    Trend: {primary['trend']}")

    except requests.HTTPError as e:
        print(f"Error: {e}")

    print("\n" + "=" * 80)


if __name__ == "__main__":
    main()